packages/web-components/src/utils/date-utils.ts

```diff
@@ -177,7 +177,8 @@ export function validate(
     component.invalidDay = false;
   }
 
-  if (month && (month < minMonths || month > maxMonths)) {
+  if ((month && (month < minMonths || month > maxMonths)) ||
+      (!month && component.invalidDay)) {
```

Comment:
Override the invalid day message if we don't have a month.

Reply:
Could you please add some tests around the logic in this file? Right now it looks like we only have a test for the leap year logic.

Reply:
Tests have been added.